portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -45,16 +51,20 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
         return;
     }
 
-    ArrayList<String> comments = new ArrayList<String>();
+    ArrayList<ArrayList<String>> commentData = new ArrayList<ArrayList<String>>();
     Iterator<Entity> iter = results.asIterable().iterator();
     for (int i = 0; i < numComments; i++) {
         Entity e = iter.next();
         String comment = (String) e.getProperty("text");
-        comments.add(comment);
+        String email = (String) e.getProperty("email");
+        ArrayList<String> data = new ArrayList<String>();
```

Comment:
Using a `Map` here instead might be clearer, so that you have JSON like:

```json
[
    {"email": "alex@foo.example", "body": "Hello!"},
    {"email": "blair@foo.example", "body": "Hi!"}
]
```